packages/seducer/src/logger.js

```diff
@@ -0,0 +1,34 @@
+/* eslint-disable no-use-before-define */
+/**
+ * Summary: Prints the current processing action via the dev console, with it's previous state and the next state after passing inside the reducer.
+ *
+ * @param {{unknown}}           prevState           The current state of the store
+ * @param {{unknown}}           nextState           The next state after being pass through the reducer
+ * @param {{ type, payload }}   action              The actions that has being dispatched by the consumer
```

Comment:
btw do we need to explicitly say `type: string, payload: any` something like that?

Reply:
I guess I can add it? is just for documentation purpose.

Reply:
Oh just for documentation not for ts?
cause I just copy-paste this file to https://bit.ly/3q4hxrw you can see some errors have been highlighted, I thought this part was for JSDoc and ts

Reply:
I got inspired by JSDoc, do you know if we can hook it up and created the TS files?

Reply:
ohhh I see what you did there, and makes sense why you were saying about `type: string, payload: any` cool, I will update those signatures.